backend/src/test/java/neuefische/capstone/bpmndiagram/BpmnDiagramServiceTest.java

```diff
@@ -0,0 +1,144 @@
+package neuefische.capstone.bpmndiagram;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.*;
+
+
+class BpmnDiagramServiceTest {
+
+    private final BpmnDiagramRepository repository = mock(BpmnDiagramRepository.class);
+    private final ServiceUtils serviceUtils = mock(ServiceUtils.class);
+    private final BpmnDiagramService service = new BpmnDiagramService(repository, serviceUtils);
+
+    @Test
+    void getAllDiagrams() {
+        //given
+        when(repository.findAll()).thenReturn(new ArrayList<>());
```

Comment:
not sure but maybe it is better to use a List of
ich zitiere : In diesem Fall ist die veränderbare ArrayList nicht optimal. Der Productioncode könnte die Liste manipulieren (was er nicht soll) und der Test wäre trotzdem grün.

Vorschlag: List.of nutzen (immutable)